### PR TITLE
[RLlib] Attention Net prep PR #1: Smaller cleanups.

### PR DESCRIPTION
**Pull request by sven1977**
The current attention net trajectory view PR (https://github.com/ray-project/ray/pull/11729) is too large (>1000 lines added).
Therefore, I'm moving smaller preparatory and cleanup changes in ~2 pre-PRs. This is the first one of these.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/agents/ddpg/ddpg_torch_policy.py [hunk 1]
```diff
@@ -274,7 +274,7 @@ def setup_late_mixins(policy, obs_space, action_space, config):
     optimizer_fn=make_ddpg_optimizers,
     validate_spaces=validate_spaces,
     before_init=before_init_fn,
-    after_init=setup_late_mixins,
+    before_loss_init=setup_late_mixins,
```
Comment (sven1977) [inline rllib/agents/ddpg/ddpg_torch_policy.py:277]: This is the more accurate kwarg to use (torch did not have a loss init step before, so this is new). The old after_init still works the exact same and thus this does not cause an API-break.

File: rllib/agents/ppo/ppo_tf_policy.py [hunk 2]
```diff
@@ -47,7 +47,12 @@ def ppo_surrogate_loss(
 
     # RNN case: Mask away 0-padded chunks at end of time axis.
     if state:
-        max_seq_len = tf.reduce_max(train_batch["seq_lens"])
+        # Derive max_seq_len from the data itself, not from the seq_lens
```
Comment (sven1977) [inline rllib/agents/ppo/ppo_tf_policy.py:50]: Prep for attention nets, where dynamic max'ing over the given sequences is not allowed.

File: rllib/evaluation/sampler.py [hunk 3]
```diff
@@ -1033,7 +1033,9 @@ def _process_observations_w_trajectory_view_api(
                 agent_id)
             episode._set_last_observation(agent_id, filtered_obs)
             episode._set_last_raw_obs(agent_id, raw_obs)
-            episode._set_last_info(agent_id, infos[env_id].get(agent_id, {}))
+            # Infos from the environment.
```
Comment (sven1977) [inline rllib/evaluation/sampler.py:1036]: Adding "infos" to the collector's, if required.

**Review by ericl — APPROVED** [review]
